src/ripple/overlay/impl/OverlayImpl.h

```diff
@@ -456,6 +456,60 @@ class OverlayImpl : public Overlay
 
     void
     sendEndpoints();
+
+private:
+    struct Stats
+    {
+
+        template <class Handler>
+        Stats (Handler const& handler, beast::insight::Collector::ptr const& collector)
+            : hook (collector->make_hook (handler))
+            , peerDisconnects (collector->make_gauge("Overlay","Peer_Disconnects")) 
+            { 
+                mcollector=collector;
+            }
+
+        beast::insight::Hook hook;
+        beast::insight::Gauge peerDisconnects;
```

Comment:
Why not add a `peer_disconnects` value to the `enum TrafficCounter::category` and merge this member into `trafficGauges`?

Reply:
I believe that enum is only for TrafficStats. Each category in that enum has a bytesIn, bytesOut, messagesIn, and messagesOut. It made sense to me to keep this one separate.